### DISCUSSION TITLE
config.json access

### DISCUSSION THREAD

**Discussion by meatalhead** [containrrr/watchtower#632] [Q&A] [answered]
Hello,

I'm running the following script for my watchtower:

```
docker run -d \
    --name watchtower \
    --net host \
    --restart=unless-stopped \
    -e TZ="Europe/London" \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -v /home/meatalhead/watchtower/config/configd.json:/config.json \
    -v /etc/timezone:/etc/timezone:ro \
    containrrr/watchtower:latest \
    --cleanup true \
    --debug true \
    --include-stopped true \
    --debug \
    --run-once \
    node_red watchtower
```

Unfortunately nothing is updating and I'm getting the following error:
`level=debug msg="Error loading authentication credentials /config.json: read /config.json: is a directory"`

I've removed the config folder from my user directory but it's recreated with the same problem. Any ideas what I'm doing wrong?

Thanks

**Comment by piksel** [comment, marked as answer]
Does `/home/meatalhead/watchtower/config/configd.json` exist as a file on your host machine? If it doesn't then Docker will assume that it's a directory and create it.

**Comment by meatalhead** [comment]
Thank you, I've removed the directory and `touch /home/meatalhead/watchtower/config/configd.json` that has helped but I'm now getting:
`"Unable to find default config file /config.json: EOF"`
`"Error loading authentication credentials /config.json: EOF"`

Does the config file need a default config copying in?

**Reply by piksel** [reply]
Well, if you don't have any credentials etc. you don't need to map the config at all. Just omit that volume from the command and it should work fine.

**Reply by simonaronsson** [reply]
If you actually want to use your docker config file, it is likely to be located at `~/.docker/config.json`. If you want to use it from the location you've specified, you'd have to copy or symlink it there first.

**Comment by meatalhead** [comment]
Thank you both, that seems to have got it working, I've just removed the link to the config.json.

In the logs I'm getting `No credentials for [container] in /config.json"`, does this matter?

**Reply by piksel** [reply]
No, it's just a warning in case of authentication failures later when it tries to fetch the image and you need to figure out why the credentials didn't work.

**Comment by meatalhead** [comment]
Thank you
